src/pages/signup/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import Input from "@/src/components/signinPage/Input";
+import styled from "styled-components";
+import SubmitBtn from "@/src/components/signinPage/SubmitBtn";
+import LogoBox from "@/src/components/signinPage/LogoBox";
+import { useForm } from "react-hook-form";
+import SocialBox from "@/src/components/signinPage/SocialBox";
+import { chechEmail, signupRequest } from "@/src/utils/Api";
+import { useRouter } from "next/router";
+
+const BackGround = styled.div`
+  width: 100%;
+  padding: 80px 0px 180px 0px;
+  margin: 0 auto;
+  background: var(--Linkbrary-bg, #f0f6ff);
+`;
+
+const Layout = styled.div`
+  width: 400px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin: 0 auto;
+  gap: 30px;
+
+  @media (max-width: 767px) {
+    width: 325px;
+  }
+`;
+
+const Form = styled.form`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  gap: 20px;
+`;
+
+function SignupPage() {
+  const router = useRouter();
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    watch,
+  } = useForm({ mode: "onBlur" });
+
+  const emailValidation = register("userEmail", {
+    required: {
+      value: true,
+      message: "이메일을 입력해 주세요",
+    },
+    pattern: {
+      value: /^\S+@\S+$/i,
+      message: "올바른 이메일 주소가 아닙니다.",
+    },
+    validate: async (value) => {
+      const res = await chechEmail(value);
+      if (!res) return "이미 존재하는 이메일입니다.";
+    },
+  });
+
+  const passwordValidation = register("userPw", {
+    required: {
+      value: true,
+      message: "비밀번호를 입력해 주세요",
+    },
+    pattern: {
+      value: /^(?=.*[a-zA-Z])(?=.*\d).{8,}$/,
+      message: "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.",
+    },
+  });
+
+  const passwordCheckValidation = register("userPwCh", {
+    required: {
+      value: true,
+      message: "비밀번호를 입력해 주세요",
+    },
+    validate: (value) => {
+      if (watch("userPw") !== value) return "비밀번호가 일치하지 않아요.";
+    },
+  });
+
+  const onSubmit = async (data: any) => {
+    const res = await signupRequest(data);
+
+    if (res) {
+      const { data } = res;
+      window.localStorage.setItem("accessToken", data.accessToken);
+      router.push("/signin");
+    }
```

Comment:
c: 사용자 인증/인가 로직은 커스텀 훅 등으로 분리해주세요. 다른 컴포넌트에서 쉽게 재사용할 수 있고 여러 인증 방식 추가에 유연하게 대응할 수 있습니다.

```suggestion
    try {
      // API 요청, 토큰 저장 등을 분리
      await signup(data);
      
      router.push("/signin");
    } catch {
      // 예외 처리
    }
```